crates/matrix-sdk/src/widget_api/driver/widget.rs

```diff
@@ -0,0 +1,78 @@
+use async_trait::async_trait;
+
+use super::handler::Result;
+use crate::widget_api::messages::capabilities;
+
+/// The Widget trait that needs to be implemented in the native client.
+/// An instance implementing this trait will be required to initialize the widget driver.
+#[async_trait]
+pub trait Widget {
```

Comment:
It could also be made as a seperate pr. But I think it does not hur here and having the functions described with documentation is definitly useful. Since this is what I envision as the final FFI bound layer. If you envision the final FFI layer differently we can of course change it.

Reply:
I think I'm ok with any traits as long as they are used somewhere in the code.